### PR TITLE
Fix Fortran bigarray tests on 64-bit Windows

### PR DESCRIPTION
**Pull request by dra27**
The function `c_filltab` in `testsuite/tests/lib-bigarray-2/bigarrfstub.c` passes the constants 8 and 6 to `caml_ba_alloc_dims`. I'm not totally clear as to why, but on 64-bit Windows those constants are passed as 32-bit constants (actually, that part makes sense - I'm less clear why this works on Linux, to be honest...).

That's the first (tiny) commit.

The Fortran test was the only place `caml_ba_alloc_dims` - the second commit just splits the test into two - one testing the C interface which is always run, and one testing the Fortran interface, if gfortran is available.

### CODE REVIEW

File: testsuite/tests/lib-bigarray-2/bigarrcstub.c [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    for (y = 0; y < DIMY; y++)
+      printf("  %6.1f", tab[x][y]);
+    printf("\n");
+  }
+}
+
+value c_filltab(value unit)
+{
+  filltab();
+  return alloc_bigarray_dims(BIGARRAY_FLOAT64 | BIGARRAY_C_LAYOUT,
```
Comment (xavierleroy) [inline testsuite/tests/lib-bigarray-2/bigarrcstub.c]: Now would be a good  time to convert to the new `caml_` runtime function names:
```suggestion
  return caml_ba_alloc_dims(BIGARRAY_FLOAT64 | BIGARRAY_C_LAYOUT,
```

File: testsuite/tests/lib-bigarray-2/bigarrfstub.c [hunk 2]
```diff
@@ -21,48 +21,11 @@ extern void filltab_(void);
...
-value c_printtab(value ba)
-{
-  printtab(Data_bigarray_val(ba));
-  return Val_unit;
-}
-
 value fortran_filltab(value unit)
 {
   filltab_();
   return alloc_bigarray_dims(BIGARRAY_FLOAT32 | BIGARRAY_FORTRAN_LAYOUT,
```
Comment (xavierleroy) [inline testsuite/tests/lib-bigarray-2/bigarrfstub.c]: Likewise:
```suggestion
  return caml_ba_alloc_dims(BIGARRAY_FLOAT32 | BIGARRAY_FORTRAN_LAYOUT,
```

**Review by xavierleroy — APPROVED** [review]
Looks good to me.  Minor suggestion below.

**Comment by xavierleroy** [comment]
> I'm not totally clear as to why, but on 64-bit Windows those constants are passed as 32-bit constants (actually, that part makes sense - I'm less clear why this works on Linux, to be honest...).

Great mysteries of the calling conventions.  On x86-64 ELF, six integer arguments can be passed in registers, but only 4 on Win64.  So, in one case, the 32-bit constants (5th and 6th arguments) are passed in integer registers, with implicit zero-extension to 64 bits.  In the other case, they are passed on the stack, probably stored as 32-bit integers in a 64-bit-wide stack slot, leaving garbage in the top 32 bits.

**Comment by dra27** [comment]
Another piece of puzzling difference to watch for in a future problem! 🙂

I just pushed a commit to go the whole hog with `caml_ba_alloc_dims` and just use `CAML_NAME_SPACE`.

**Pull request merged**